resources/resource-extensbility/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,432 @@
+1. Users who are familiar with openAPI schema format and are proficient in defining the schema in YAML/JSON format
+
+Hence, we need to support both the schema formats in Radius to support both user groups.
+
+*User experience*
+
+1. Deb is a platform engineer at Contoso open to use typespec and wants to author a custom resource Plaid in Radius
+
+    Pre-requisites: User has typespec compiler/CLI installed
+
+    1. Deb uses the radCLI to scaffold a template typespec definition
+
+        ```bash
+        rad resource-provider init contoso --template-kind typespec
+        ```
+        A sample plaid.tsp file and manifest files are created in the application folder with all the typespec dependencies installed.
+
+        Question :Can we build on top of `tsp init` which is used to set up a typespec project with a sample template? The other option is to use `rad resource-provider new` to scaffold a template typespec definition and the user takes care of the typespec dependencies and setup which might be a confusing.
+
+    1. Deb authors the schema for plaid in typespec format
+        
+        ```
+        import "@typespec/http";
+        import "@typespec/rest";
+        import "@typespec/openapi3";
+        import "@typespec/versioning";
+
+        using TypeSpec.Http;
+        using TypeSpec.Rest;
+        using TypeSpec.Versioning;
+
+        @service({title: "Contoso"})
+        @server("http://contoso.plaid.com", "endpoint for plaid service")
+        @versioned(Versions)
+        namespace Contoso;
+
+        @route("/planes/radius/Contoso.Messaging")
+        namespace plaid {
+            model plaid {
+                name:string;
+                description:string;
+                queue:string;
+            }
+        }
+
+        ```
+
+    1. VSCode Type spec extension warns Deb on the errors and warnings as he types
+     
+        ![alt text](images/errors.png)
+
+    1. Deb uses intellisense to autocomplete the schema and properties
+
+        ![alt text](images/intellisense.png)
+
+    1. Deb chooses a version for the schema
+
+        ```
+        import "@typespec/http";
+        import "@typespec/rest";
+        import "@typespec/openapi3";
+        import "@typespec/versioning";
+
+        using TypeSpec.Http;
+        using TypeSpec.Rest;
+        using TypeSpec.Versioning;
+
+        @service({title: "Contoso"})
+        @server("http://contoso.plaid.com", "endpoint for plaid service")
+        @versioned(Versions)
+        namespace Contoso;
+
+        @route("/planes/radius/Contoso.Messaging")
+        namespace plaid {
+            model plaid {
+                name:string;
+                description:string;
+                queue:string;
+            }
+        }
+
+        enum Versions {
+            v1,
+        }
+        ```
+   1. Deb registers the schema in Radius
+    
+        ```bash
+        rad resource-provider register contoso --template-kind typespec -e myenv -v v1
```

Comment:
nit: I think we'll probably register all versions as a unit, rather than each separately.